src/snapred/backend/recipe/algorithm/ReductionAlgorithm.py

```diff
@@ -1,231 +1,126 @@
-import json
+from typing import Dict
 
-import numpy as np
-from mantid.api import AlgorithmFactory, PythonAlgorithm, mtd
+from mantid.api import AlgorithmFactory, ITableWorkspaceProperty, MatrixWorkspaceProperty, PropertyMode, PythonAlgorithm
 from mantid.kernel import Direction
+from numpy import nan
 
 from snapred.backend.dao.ingredients import ReductionIngredients
 from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
 
 
 #######################################################
 # ATTENTION: Could be replaced by alignAndFocusPowder #
-# please confirm that attenuation correction before  #
+# please confirm that attenuation correction before   #
 # and after is equivalent                             #
 #######################################################
 class ReductionAlgorithm(PythonAlgorithm):
+    """
+    NOTE: this was originally written to work with a list of grouping workspaces,
```

Comment:
So one catch with this is we may need to split the algo if it only applied to one group.  The reason it was for a set of groups was there were "cached" steps reused for each group.